pcdsdevices/epics_motor.py

```diff
@@ -323,12 +337,75 @@ def egu(self) -> str:
         """
         return self._egu
 
-    def _cache_egu(self, value: str, **kwargs):
+    def _cache_egu(self, value: str, **kwargs) -> None:
         """
         Cache the value of EGU for later use in the egu property.
         """
         self._egu = value
 
+    @classmethod
+    def _motion_error_filter(cls, record: logging.LogRecord) -> bool:
+        """
+        A filter for removing spammy end of move logs.
+
+        This removes the cross-session error spam issues.
+        This should be added to the ophyd.objects filter exactly once
+        per session.
+        """
+        try:
+            name = record.ophyd_object_name
+        except AttributeError:
+            logger.debug('logging filter applied to wrong logger')
+            return True
+        try:
+            filt = cls._motion_error_log_filters[name]
+        except KeyError:
+            logger.debug(f'logging filter not found for {name}')
```

Comment:
Aren't we supposed to use %s syntax instead for logger messages?

Reply:
%s in logger messages has two advantages that I know of:
- Allows programs to filter out repeated messages better, since they can be grouped by the original text, provided we intend to use such filters.
- Is slightly faster when a log message is not logged, since the substitution operation can be skipped.

And in my head, one disadvantage:
- The code is harder to read since we are being encouraged to use different substitution strategies in different places.

I'm not interested in putting %s back into my code for any reason and I wish python would support more formatting options in log messages.

Reply:
I lean toward `%`-logger-formatting as good practice for the reasons Zach mentioned, plus the fact that formatting bugs can't cause unexpected exceptions. I'd consider that the primary advantage, since a `logger.debug` causing a crash is a facepalm moment for me (I frequently mess them up, if I'm being honest).

I second the wish for better formatting options, and also wouldn't reject what is proposed here. It would be ideal to get coverage on such lines.

Reply:
I'll probably either add a test or remove these debug statements outright because it's a bit much to bother. I added these during a testing loop where I couldn't figure out why the filter wasn't filtering.